### PR TITLE
feat(877): highest sale for series

### LINKED ISSUE #877: Add highest sale
Would be nice to see the highest sale for 
- [x] collections
- [ ] series view

Add a column with the highest sale to the series insight. 

#### side notes
But has an issue that how we know it's not a washtrade?

Could be joined with somewhere on top of 
- #1818
### Ref
- #2801

It's already in collections meanwhile
image

### PR DESCRIPTION
**Pull request by KngZhi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #877
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [x] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer?target=DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc&usdamount=100&donation=true)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

image

### CODE REVIEW

**Comment by yangwao** [comment]
Why are there 0 on screenshot for the highest sale?

**Comment by KngZhi** [comment]
It seems the Rubick hasn't deployed?

from the local graphQL of Rubick, I can see the data. But on Kodadot graphQL, it would occur error.

Screen Shot 2022-04-29 at 3 22 09 PM

Screen Shot 2022-04-29 at 3 23 02 PM

**Review by roiLeo — COMMENTED** [review]
- missing `highestSale` type in RowSeries
- missing `highestSale` in SeriesTable columns

I think subsquid takes 2-3 days to deploy @vikiival can confirm this?

**Comment by roiLeo** [comment]
> oh, I get it, you mean the `protected columns: Column[]`, from what I see and check with the file, that part code seems no longer used.
> 
> I think the original idea for that columns is to use `v-for`, but now the column gets more complicated, thus deprecated.

If not used I would remove it as it can be misleading

**Comment by KngZhi** [comment]
Oh, I would remove that, and update with the `main` branch.

File: langDir/en.json [hunk 1]
```diff
@@ -584,6 +584,7 @@
     "buys": "Buys",
     "collected": "Collected",
     "floorprice": "Floor price",
+    "highestSale": "Highest Sale",
```
Comment (vikiival) [inline langDir/en.json:587]: I think this won't work yet, fixin' stuff on rubick@006

File: queries/rmrk/subsquid/seriesInsightList.graphql [hunk 2]
```diff
@@ -9,6 +9,7 @@ query seriesInsightList(
   ) {
     id
     floorPrice
+    highestSale
```
Comment (vikiival) [inline queries/rmrk/subsquid/seriesInsightList.graphql:12]: I think this won't work yet, fixin' stuff on rubick@006

**Comment by yangwao** [comment]
At https://deploy-preview-2915--koda-nuxt.netlify.app/series-insight/

```
{"errors":[{"message":"Cannot query field \"highestSale\" on type \"Series\".","extensions":{"code":"GRAPHQL_VALIDATION_FAILED","exception":{"stacktrace":["GraphQLError: Cannot query field \"highestSale\" on type \"Series\".","    at Object.Field (/squid/node_modules/graphql/validation/rules/FieldsOnCorrectTypeRule.js:48:31)","    at Object.enter (/squid/node_modules/graphql/language/visitor.js:323:29)","    at Object.enter (/squid/node_modules/graphql/utilities/TypeInfo.js:370:25)","    at visit (/squid/node_modules/graphql/language/visitor.js:243:26)","    at validate (/squid/node_modules/graphql/validation/validate.js:69:24)","    at validate (/squid/node_modules/apollo-server-core/dist/requestPipeline.js:186:39)","    at processGraphQLRequest (/squid/node_modules/apollo-server-core/dist/requestPipeline.js:98:34)","    at runMicrotasks (<anonymous>)","    at processTicksAndRejections (node:internal/process/task_queues:96:5)","    at async processHTTPRequest (/squid/node_modules/apollo-server-core/dist/runHttpQuery.js:187:30)"]}}}]}
```

image

**Comment by KngZhi** [comment]
Screen Shot 2022-04-30 at 4 49 02 PM

Using Rubick-005 right now, maybe wait for some time.

**Comment by vikiival** [comment]
@KngZhi

rubick@006 is ready to use please chceck discord garage group
Screenshot 2022-04-30 at 15 20 46

**Comment by yangwao** [comment]
@KngZhi let us know when it's ready for testing

**Comment by KngZhi** [comment]
[db migration seems wrong, need redeploy](https://github.com/kodadot/rubick/issues/61#issuecomment-1115895365)

**Comment by vikiival** [comment]
Should be up in a couple of minutes :)

**Review by roiLeo — COMMENTED** [review]
CuratedList in landing is not working?
Capture d’écran 2022-05-03 à 5 25 47 PM

https://deploy-preview-2915--koda-nuxt.netlify.app/series-insight?sort=-volume&rows=20
is showing 5 lines while https://deploy-preview-2915--koda-nuxt.netlify.app/series-insight?sort=-volume&rows=50 is showing 3 lines (see server response)

Capture d’écran 2022-05-03 à 5 32 34 PM

**Comment by KngZhi** [comment]
> CuratedList in landing is not working

it has to relate to `nuxt.config.js`, I just updated the endpoint. if changed to `005` it would be fine, but the `highest_sale` would not show up..

> edit: is migration in progress? How long does it take?

Yes. The data is migrating.

File: nuxt.config.js [hunk 3]
```diff
@@ -258,10 +258,10 @@ export default defineNuxtConfig({
       ...defineApolloConfig(),
       subsquid: toApolloEndpoint(
         process.env.SUBSQUID_ENDPOINT ||
-          'https://app.gc.subsquid.io/beta/rubick/005/graphql'
+          'https://app.gc.subsquid.io/beta/rubick/006/graphql'
       ),
       legacysquid: toApolloEndpoint(
-        'https://app.gc.subsquid.io/beta/rubick/004/graphql'
+        'https://app.gc.subsquid.io/beta/rubick/005/graphql'
```
Comment (vikiival) [inline nuxt.config.js:264]: nop :) leave this at 004

**Comment by vikiival** [comment]
> How long does it take

Couple of hrs until finish :)

Screenshot 2022-05-04 at 12 30 29
?

**Review by roiLeo — APPROVED** [review]
✅

**Comment by roiLeo** [comment]
pay 250

**Comment by yangwao** [comment]
pay 250 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $250 @ 124 USD/KSM ~ 2.016 $KSM
🧗 [DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc](https://kusama.subscan.io/account/DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc)
🔗 [0x13866875e98e6dfca6b958ca4cfe230f5b2a743a4bbc549ab10e1357fe9e97e9](https://kusama.subscan.io/extrinsic/0x13866875e98e6dfca6b958ca4cfe230f5b2a743a4bbc549ab10e1357fe9e97e9)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**